File: updated_code/code/java/009-palindrome-number.java
/** Package: com.example.palindromechecker */

import java.io.*;

class Solution {
    public boolean isPalindrome(int x) {
        // Check if x is negative
        if (x < 0) {
            return false;
        }

        int reversed = 0;
        // Declare 'original' as final
        final int original = x;
        while (x > 0) {
            // Declare 'remainder' as final
            final int remainder = x % 10;
            reversed = reversed * 10 + remainder;
            x = x / 10;
        }
        // Return the result
        return original == reversed;
    }
}

File: updated_code/code/python/009-palindrome-number.py
class Solution:
    """
    Description: Given an integer `x`, return `true` _if_`x` _is a_ _ **palindrome**_ _,
    and_`false` _otherwise_.

    Follow-up: Could you solve it without converting the integer to a string?
    """
    def is_palindrome(self, x: int) -> bool:
        """
        Check if the integer is a palindrome.

        Args:
        x (int): The input integer.

        Returns:
        bool: True if the integer is a palindrome, False otherwise.
        """
        if x < 0:  # Negative numbers are not palindromes
            return False
        reversed_x = 0
        original_x = x
        while x != 0:
            remainder = x % 10
            reversed_x = reversed_x * 10 + remainder
            x = x // 10
        return original_x == reversed_x


# Add a newline at the end of the file